### PR TITLE
Support for multiple --namespace flag

### PR DESCRIPTION
**Pull request by uesyn**
This PR enables to watch pod logs across namespaces with multiple `--namespace` flag.
I read #73, and I think that this feature is needed.

### CODE REVIEW

**Comment by uesyn** [comment]
rebased to resolve conflict.

File: cmd/cli.go [hunk 1]
```diff
@@ -281,6 +281,20 @@ func parseConfig(args []string) (*stern.Config, error) {
...
+			nset[n] = struct{}{}
+		}
+		for n := range nset {
+			if n == "" {
+				continue
+			}
+			namespaces = append(namespaces, n)
+		}
+	}
+
```
Comment (superbrothers) [inline cmd/cli.go:297]: Please add a comment that makes the array unique.

Comment (superbrothers) [reply cmd/cli.go:297]: The below may be easier to read.

```suggestion
	namespaces := []string{}
	if opts.namespaces != nil {
		m := make(map[string]struct{})
		for _, namespace := range opts.namespaces {
			if namespace == "" {
				continue
			}

			if _, ok := m[namespace]; !ok {
				m[namespace] = struct{}{}
				namespaces = append(namespaces, namespace)
			}
		}
	}


```

Comment (uesyn) [reply cmd/cli.go]: 106178b 656cecc
I corrected the points that were pointed out!

File: stern/stern.go [hunk 2]
```diff
@@ -60,32 +60,59 @@ func Run(ctx context.Context, config *Config) error {
...
-		config.ExcludeContainerQuery,
-		config.InitContainers,
-		config.ContainerState,
-		config.LabelSelector,
-		config.FieldSelector)
-	if err != nil {
-		return errors.Wrap(err, "failed to set up watch")
+	added := make(chan *Target)
+	removed := make(chan *Target)
+	errCh := make(chan error)
```
Comment (superbrothers) [inline stern/stern.go:80]: These channels may need to be closed.

**Comment by superbrothers** [comment]
If multiple namespaces are specified, it is more helpful to output the namespace as in the case of `--all-namespaces`.

```
$ dist/stern . --all-namespaces --since 1s
+ kube-system kube-proxy-j7dcb › kube-proxy
+ kube-system coredns-74ff55c5b-h2nb8 › coredns
+ kube-system coredns-74ff55c5b-sjqn9 › coredns
+ default nginx-74dcc88b89-7gnjl › nginx-unprivileged
+ kube-system kube-apiserver-kind-control-plane › kube-apiserver
+ kube-system etcd-kind-control-plane › etcd
+ kube-system kube-controller-manager-kind-control-plane › kube-controller-manager
+ kube-system kube-scheduler-kind-control-plane › kube-scheduler
+ kube-system kindnet-qlx4g › kindnet-cni
+ local-path-storage local-path-provisioner-78776bfc44-4sfw6 › local-path-provisioner
^C
$ dist/stern . --namespace default,kube-system --since 1s
+ nginx-74dcc88b89-7gnjl › nginx-unprivileged
+ kube-proxy-j7dcb › kube-proxy
+ coredns-74ff55c5b-h2nb8 › coredns
+ coredns-74ff55c5b-sjqn9 › coredns
+ kube-scheduler-kind-control-plane › kube-scheduler
+ kindnet-qlx4g › kindnet-cni
+ kube-apiserver-kind-control-plane › kube-apiserver
+ etcd-kind-control-plane › etcd
+ kube-controller-manager-kind-control-plane › kube-controller-manager
^C
```

The following changes probably need to be made:

```diff
diff --git a/stern/stern.go b/stern/stern.go
index 0098cd9..70b1782 100644
--- a/stern/stern.go
+++ b/stern/stern.go
@@ -133,7 +136,7 @@ func Run(ctx context.Context, config *Config) error {
                                SinceSeconds: int64(config.Since.Seconds()),
                                Exclude:      config.Exclude,
                                Include:      config.Include,
-                               Namespace:    config.AllNamespaces,
+                               Namespace:    config.AllNamespaces || len(namespaces) > 1,
                                TailLines:    config.TailLines,
                        })
                        setTail(targetID, tail)
```

File: stern/stern.go (hunk 2 above)
Comment (uesyn) [reply stern/stern.go:80]: Fixed it! 1774436

**Comment by uesyn** [comment]
I couldn't notice that. Thanks! Fixed it.

**Comment by superbrothers** [comment]
Could you squash the commits? I'll merge this PR after you did that.

**Review by floryut — APPROVED** [review]

**Comment by floryut** [comment]
> Could you squash the commits? I'll merge this PR after you did that.

Squashes, I'm merging it then :)

**Pull request merged**